### PR TITLE
Allows for better UX when interacting with the product store selection on the product show page

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
This just adds an overlay / loader to demonstrate that something is happening

https://developertown.visualstudio.com/SIL%20International%20Scriptoria/_workitems/edit/34492

### CODE REVIEW

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/components/loaders/overlay-loader.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import RectLoader from './rect-loader';
+
+export default function OverlayLoader(props: any) {
+  return (
+    <div
+      data-test-overlay-loader
+      className='flex-row flex-grow justify-content-conter align-items-center'
+      {...props}
+      style={{
+        opacity: 0.5,
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/components/loaders/overlay-loader.tsx]: inline styles are cool in react, right?

these aren't reusable, so here they be

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/components/loaders/overlay-loader.tsx]: maybe they would be if we had a legit style compiler though.

one day

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/overview/products/selection-manager/store-selection/index.tsx [hunk 2]
```diff
@@ -23,53 +25,64 @@ interface INeededProps {
...
+      // loading state again
+      // setLoading(false);
+    },
+    [selected]
+  );
 
-    const { name } = attributesFor(productDefinition);
+  const { name } = attributesFor(productDefinition);
 
-    const selectProps = {
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/overview/products/selection-manager/store-selection/index.tsx]: these props just moved inline

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/overview/products/selection-manager/store-selection/index.tsx [hunk 3]
```diff
@@ -23,53 +25,64 @@ interface INeededProps {
...
+  return (
+    <Modal
+      data-test-project-product-store-select-modal
+      open={true}
+      className='medium p-relative'
+      style={{ zIndex: 1 }}
+      closeIcon={<CloseIcon className='close-modal' />}
+      onClose={onStoreCancel}
+    >
+      {isLoading && <OverlayLoader style={{ zIndex: 2 }} />}
```
Comment (NullVoxPopuli) [inline source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/projects/show/overview/products/selection-manager/store-selection/index.tsx]: this, along with the Modal's style addition is the main change in this file.

File: source/SIL.AppBuilder.Portal.Frontend/src/ui/components/loaders/overlay-loader.tsx (hunk 1 above)
Comment (dalinarkholin) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/components/loaders/overlay-loader.tsx]: They still clutter the jsx, and are meaningless unless you are styling. Which is why I keep them in a local stylesheet as a sibling. :)

Comment (dalinarkholin) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/components/loaders/overlay-loader.tsx]: I will let it pass THIS time ;)

Comment (NullVoxPopuli) [reply source/SIL.AppBuilder.Portal.Frontend/src/ui/components/loaders/overlay-loader.tsx]: they do. This was *pure* laziness

**Pull request merged**